src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -258,20 +262,50 @@ private function checkClass($class, $file = null)
                     }
                 }
 
-                // Method from a trait
-                if ($method->getFilename() !== $refl->getFileName()) {
```

Comment:
This should be kept imo, I think we also don't want methods from traits being checked against their own `@param` annotations.

Reply:
worth a new test case :)
we also want to report implementation provided by traits that don't provide the argument declared by parents

Reply:
> we also want to report implementations provided by traits that don't provide the argument declared by parents

👍, you were right about this change.

> and also to report missing arguments declared by @param on traits

Thinking of it again, I prefered not parsing `@param` in traits directly since PHP allows a full overwriting of their methods, instead their methods are checked as if they belong to the sub classes using them.